lamindb/_record.py

```diff
@@ -168,68 +169,98 @@ def suggest_records_with_similar_names(
-                if hasattr(record, "clean_fields"):
-                    record.clean_fields()
-                else:
-                    record._Model__clean_fields()
-            except DjangoValidationError as e:
-                message = _format_django_validation_error(record, e)
-                raise FieldValidationError(message) from e
-    elif len(args) != len(record._meta.concrete_fields):
-        raise ValueError("please provide keyword arguments, not plain arguments")
-    else:
-        # object is loaded from DB (**kwargs could be omitted below, I believe)
-        super(BasicRecord, record).__init__(*args, **kwargs)
-        _store_record_old_name(record)
-        _store_record_old_key(record)
+                        version_comment = ""
+                        existing_record = exact_match
+                    if existing_record is not None:
+                        logger.important(
+                            f"returning existing {record.__class__.__name__} record with same"
+                            f" {name_field}{version_comment}: '{kwargs[name_field]}'"
+                        )
+                        init_self_from_db(record, existing_record)
+                        update_attributes(record, kwargs)
+                        return None
+            super(BasicRecord, record).__init__(**kwargs)
+            if isinstance(record, ValidateFields):
+                # this will trigger validation against django validators
+                try:
+                    if hasattr(record, "clean_fields"):
+                        record.clean_fields()
+                    else:
+                        record._Model__clean_fields()
+                except DjangoValidationError as e:
+                    message = _format_django_validation_error(record, e)
+                    raise FieldValidationError(message) from e
+        elif len(args) != len(record._meta.concrete_fields):
+            valid_field_names = [
+                f.name
+                for f in record._meta.concrete_fields
+                if not f.name.startswith("_")
+                and f.name
+                not in (
+                    "id",
+                    "uid",
+                    "created_at",
+                    "updated_at",
+                    "created_by",
+                    "run",
+                    "space",
+                    "synonyms",
+                )
+            ]
+            example = record.__class__.__name__
+            sys.exit(
+                f"Don't pass a string directly. Use: {example}(name='...')\nValid fields: {valid_field_names}"
+            )
+        else:
+            # object is loaded from DB (**kwargs could be omitted below, I believe)
+            super(BasicRecord, record).__init__(*args, **kwargs)
+            _store_record_old_name(record)
+            _store_record_old_key(record)
+    except (TypeError, AttributeError) as e:
+        if "missing required argument" in str(
```

Comment:
I think this is too layered. The "missing required argument" error message is already fine-tuned to the user and exits with a ValidationError (I'm pretty sure). So, I don't even get why this would occur under TypeError, AttributeError 🤔

Reply:
Btw, every core lamindb Record class already prints the valid keyword arguments.

For instance here:

<img width="731" alt="image" src="https://github.com/user-attachments/assets/945f08b1-27bf-4288-886c-93e55191405c" />

But same for artifact, collection, transform etc. 

So this whole effort here is double.

This being said: the validation in lamindb constructors is done manually rather than based on the overloaded signature; so I imagine this could be simplified.